### DISCUSSION TITLE
Nuxt 3. Conditionally fetching data

### DISCUSSION THREAD

**Discussion by YuLogun** [nuxt/nuxt#17252] [Questions]
Hi! 
I wonder if it's possible to fetch data conditionally in global await?

```
<script setup>
...
const test = ref([]);
if (myCondition.value) {
    const {
        data
    } = await useAsyncData('test', () => $fetch('/my/api/call'));
    test.value = data.value;
}
</script>
```

If I write it this way, test value is not reactive and is not updated in the template

**Comment by YuLogun** [comment]
later found this similar issue https://github.com/nuxt/framework/discussions/5992
still, if anyone has any other ideas, they would be really welcome..

**Reply by Flanker72** [reply]
As a suggestion for your code example. Try to make 'test.value = [...data.value]' instead: destruction should produce new array and trigger ref update.

**Reply by YuLogun** [reply]
Unfortunately, this doesn't work.
Provided that `myCondition` is reactive (updated the initial code example condition), the following would work:

```
const test = ref([]);
watch(() => myCondition.value, async (condition) => {
    if (condition) {
        const { data } = await useAsyncData('test', () => $fetch('/my/api/call'));
        test.value = data.value;
    }
});
```
